src/pages/index.js

```diff
@@ -54,26 +124,77 @@ addButton.addEventListener("click", () => {
   addNewCardFormValidator.disableSubmitButton();
   addNewCardModal.open();
 });
+updateAvatarIcon.addEventListener("click", () => {
+  updateAvatarFormValidator.disableSubmitButton();
+  updateAvatarModal.open();
+});
 
 export const userInfo = new UserInfo({
   userNameSelector: ".profile__heading",
   userOccupationSelector: ".profile__sub-heading",
+  profileAvatarSelector: ".profile__avatar",
 });
 
 export function updateProfile(profileObject) {
   const newName = profileObject["input-name"];
   const newOccupation = profileObject["input-description"];
+  console.log(profileObject);
+  profileSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfile({
+      name: newName,
+      occupation: newOccupation,
+    })
+    .finally(() => {
+      profileSubmitButton.textContent = "Save";
+    });
   userInfo.setUserInfo(newName, newOccupation);
   editProfileFormValidator.disableSubmitButton();
   editProfileModal.close();
 }
 
+export function updateProfilePicture(profileObject) {
+  const newPhotoLink = profileObject["input-url"];
+  console.log(profileObject);
+  changeAvatarSubmitButton.textContent = "Saving...";
+  api.updateAvatar(newPhotoLink).then((response) => {
+    userInfo.setUserAvatar(response.avatar);
+    changeAvatarSubmitButton.textContent = "Save";
+  });
+  updateAvatarFormValidator.disableSubmitButton();
+  updateAvatarModal.close();
+}
+
 export function addNewCard(cardData) {
   const newCard = {
     name: cardData["input-place"],
     link: cardData["input-url"],
   };
-  cardList.addItem(createCard(newCard));
+  addCardSubmitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```